geth/api/backend_txqueue_test.go

```diff
@@ -391,28 +391,26 @@ func (s *BackendTestSuite) TestCompleteMultipleQueuedTransactions() {
 
 		// complete
 		results := s.backend.CompleteTransactions(string(updatedTxIDStrings), TestConfig.Account1.Password)
-		if len(results) != (testTxCount+1) || results["invalid-tx-id"].Error.Error() != "transaction hash not found" {
-			s.Fail(fmt.Sprintf("cannot complete txs: %v", results))
-			return
-		}
+		require.Len(results, testTxCount+1)
+		require.EqualError(results["invalid-tx-id"].Error, "transaction hash not found")
+
 		for txID, txResult := range results {
 			if txResult.Error != nil && txID != "invalid-tx-id" {
-				s.Fail(fmt.Sprintf("invalid error for %s", txID))
-				return
+				require.Fail(fmt.Sprintf("invalid error for %s", txID))
 			}
+
 			if txResult.Hash.Hex() == "0x0000000000000000000000000000000000000000000000000000000000000000" && txID != "invalid-tx-id" {
-				s.Fail(fmt.Sprintf("invalid hash (expected non empty hash): %s", txID))
-				return
+				require.Fail(fmt.Sprintf("invalid hash (expected non empty hash): %s", txID))
 			}
 
 			if txResult.Hash.Hex() != "0x0000000000000000000000000000000000000000000000000000000000000000" {
-				log.Info("transaction complete", "URL", "https://rinkeby.etherscan.io/tx/%s"+txResult.Hash.Hex())
+				log.Info("transaction complete", "URL", "https://ropsten.etherscan.io/tx/"+txResult.Hash.Hex())
```

Comment:
Why this condition has only log statements? If it's not testing anything, what's the purpose?

Reply:
It's just for debugging purposes to ensure the transaction has really been sent from a public service.

Reply:
Hmm, it doesn't seem good approach to me anyway.
Concatenating hash to const URL doesn't ensure anything. If there is any check that could be made on hash hex value, we should put into the test code.

Reply:
Nope, it's for manual checking. Anyway, it's never been useful to me so far so I'll remove it.